eclass/qt5-build.eclass

```diff
@@ -58,6 +58,27 @@ esac
 # for tests you should proceed with setting VIRTUALX_REQUIRED=test.
 : ${VIRTUALX_REQUIRED:=manual}
 
+# @ECLASS-VARIABLE: QT5_DOCS
+# @DESCRIPTION:
+# List of docs tarballs that belong to a package. By default set to false.
+: ${QT5_DOCS:=false}
+
+# @ECLASS-VARIABLE: QT5_DOCSMAP
+# @DESCRIPTION:
+# IMPORTANT: Don't forget to update on version bumps!
+# Key:Value pairs of known Qt5 documentation tarballs where key=<version> and
+# value=<upstream timestamp>, sorted by <version>, descending.
+QT5_DOCSMAP=(
```

Comment:
you can probably use an associative array

```
declare -A QT5_DOCSMAP=(
    [version]=timestamp
    ...
)
```

(can also be declared `readonly` after init)

Also, isn't this an internal variable? why documenting it as `@ECLASS-VARIABLE`? I think a normal comment should be enough.

Reply:
The problem with the associative array is ordering; so we 'break' live ebuilds because they won't use 'latest available per branch' anymore but whatever hash turns out first.

Reply:
uhm, I wasn't thinking about live ebuilds at all... it sounds like they could be a headache either way... even in your implementation, whenever you add a new entry to this array (e.g. on a version bump), do you have to remanifest all live ebuilds? I'm not sure I like this.

Reply:
With an associative array, we could have separate entries for each 5.x.9999 branch, and bump them separately if desired. This would lead to less coupling.

Reply:
True, re-manifesting the live ebuilds on version bumps would be the consequence I did not think about myself yet. On the plus side, version bumps don't happen that often. I don't think there is another way to supply live ebuild users with the docs though?

Other solutions? We could split them out and mirror each Qt package in `app-doc/*`, that creates yet another haskell/acct-user/acct-group/virtual situation and we would need to hope for https://bugs.gentoo.org/691798 to be fixed soon.

Or we just keep the one true `dev-qt/qt-docs` ebuild and just add USE flags for all our Qt packages.

Reply:
(the more I think of it, the more I just want to keep it all in qt-docs)

Reply:
> Yes, but on the other hand, would we ever not bump those when we have a new Qt version bump? It would even be the same manifest run when done in Qt overlay. Fatal, of course, if missed.

Probably. But the point is that we wouldn't *have* to, e.g. we could bump those only at each new minor Qt version, not at each patch-level version bump. I'm not saying we have to do that, it just gives us the option to do it. Less coupling is good.